### PR TITLE
feat(pharos): add layout component

### PR DESCRIPTION
**Pull request by Niznikr**
**This change:** (check at least one)

- [x] Adds a new feature
- [ ] Fixes a bug
- [ ] Improves maintainability
- [ ] Improves documentation
- [ ] Is a release activity

**Is this a breaking change?** (check one)

- [ ] Yes
- [x] No

**Is the:** (complete all)

- [x] Title of this pull request clear, concise, and indicative of the issue number it addresses, if any?
- [x] Test suite(s) passing?
- [x] Code coverage maximal?
- [x] Changeset added?

**What does this change address?**
Add a layout component for our various page layouts used on the platform. The three layouts include:

- `1 column`: margins range from 1.25rem to 9rem with fixed-width content
- `1 column w/sidenav`: margin is 1.5rem on mobile, otherwise 3.5rem with fluid content
- `2 column`: margin is 1.5rem on mobile, otherwise 3.5rem with fluid content

The layouts should support our defined breakpoints:

- `small`: 360px
- `medium`: 768px
- `large`: 1056px
- `xlarge`: 1360px
- `max`: 1584px

**How does this change work?**

- Add design tokens for breakpoints and grid gutter
- Add layout component which applies a 12-column grid for a specified layout
- Add Sass mixins for establishing grid styles for the breakpoints
- Add Sass maps for storing column and margin design decisions
- Update example Storybook pages to utilize the component

**Additional context**
CSS variables cannot currently be used for media queries so I opted to use the Sass tokens.

### CODE REVIEW

**Review by daneah — COMMENTED** [review]
Overall I think I enjoy the DX of this. This component feels geared pretty specifically toward page-level layouts, and as such I wonder if it might be better to name it `<pharos-layout>` or similar to avoid conveying that it should be used for granular gridding needs.

File: packages/pharos/src/pages/item-detail/react/item-detail.pages.stories.tsx [hunk 1]
```diff
@@ -26,53 +27,55 @@ export default {
...
-          <div className="item-detail-page__container--viewer"></div>
-          <div className="item-detail-page__container--collections">
-            <ItemCarousel />
-            <CollectionCarousel />
+    <main>
+      <PharosGrid
+        layout="2-col"
+        rows="max-content 1fr"
+        className="item-detail-page__container--main-content"
+      >
```
[the remark this answers is not in the document]
Comment (Niznikr) [reply packages/pharos/src/pages/item-detail/react/item-detail.pages.stories.tsx:35]: For that situation you could either use the `areas` prop to utilize `grid-areas` to structure things or set the element's `grid-column` via CSS to span all 12 columns.

**Comment by Niznikr** [comment]
Yeah `<pharos-layout>` might work better! Will see how others feel about that.

**Review by SMQuazi — APPROVED** [review]

File: packages/pharos/src/pages/item-detail/react/item-detail.pages.stories.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (daneah) [reply packages/pharos/src/pages/item-detail/react/item-detail.pages.stories.tsx:35]: Ah I can see it in my head now. Thanks for clarifying.

File: packages/pharos/src/components/layout/PharosLayout.react.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+export const gridColumns = () => {
+  const arr = [];
+  for (let i = 1; i <= 12; i++) {
+    arr.push(<div className="layout-example__column">{i}</div>);
+  }
+  return arr;
+};
+
+# Grid
```
Comment (daneah) [inline packages/pharos/src/components/layout/PharosLayout.react.stories.mdx:55]:
```suggestion
# Layout
```

**Review by daneah — APPROVED** [review]

**Pull request merged**